java/src/Ice/src/main/java/com/zeroc/Ice/OutputStream.java

```diff
@@ -9,138 +9,61 @@
-        final Instance instance = communicator.getInstance();
-        initialize(instance, instance.defaultsAndOverrides().defaultEncoding, direct);
+    public OutputStream(EncodingVersion encoding, boolean direct) {
+        this(encoding, FormatType.CompactFormat, direct);
     }
 
     /**
-     * This constructor uses the given communicator and encoding version.
+     * Construct a new instance of the OutputStream class. The output stream is initially empty, and
+     * uses the specified encoding, and class format.
      *
-     * @param communicator The communicator to use when initializing the stream.
-     * @param encoding The desired Ice encoding version.
-     */
-    public OutputStream(Communicator communicator, EncodingVersion encoding) {
-        assert (communicator != null);
-        final Instance instance = communicator.getInstance();
-        initialize(instance, encoding, instance.cacheMessageBuffers() > 1);
-    }
-
-    /**
-     * This constructor uses the given communicator and encoding version.
-     *
-     * @param communicator The communicator to use when initializing the stream.
-     * @param encoding The desired Ice encoding version.
+     * @param encoding The encoding version to use.
+     * @param format The format to use for class encoding.
      * @param direct Indicates whether to use a direct buffer.
      */
-    public OutputStream(Communicator communicator, EncodingVersion encoding, boolean direct) {
-        assert (communicator != null);
-        final Instance instance = communicator.getInstance();
-        initialize(instance, encoding, direct);
-    }
-
-    public OutputStream(Instance instance, EncodingVersion encoding) {
-        initialize(instance, encoding, instance.cacheMessageBuffers() > 1);
-    }
-
-    public OutputStream(Instance instance, EncodingVersion encoding, boolean direct) {
-        initialize(instance, encoding, direct);
-    }
-
-    public OutputStream(Instance instance, EncodingVersion encoding, Buffer buf, boolean adopt) {
-        initialize(instance, encoding, new Buffer(buf, adopt));
-    }
-
-    /**
-     * Initializes the stream to use the communicator's default encoding version and class encoding
-     * format.
-     *
-     * @param communicator The communicator to use when initializing the stream.
-     */
-    public void initialize(Communicator communicator) {
-        assert (communicator != null);
-        final Instance instance = communicator.getInstance();
-        initialize(
-                instance,
-                instance.defaultsAndOverrides().defaultEncoding,
-                instance.cacheMessageBuffers() > 1);
+    public OutputStream(EncodingVersion encoding, FormatType format, boolean direct) {
+        // The 1.0 encoding doesn't use the class format type, but we still have to set it in case
+        // the stream reads and 1.1 encapsulation, in which case it would use the format type set
+        // in the stream.
```

Comment:
Seems like a typo.
```suggestion
        // The 1.0 encoding doesn't use the class format type, but we still have to set it in case
        // the stream reads an 1.1 encapsulation, in which case it would use the format type set
        // in the stream.
```